### PR TITLE
[Update] Anime Planet Presence

### PR DESCRIPTION
**Pull request by SlowLife1661**
## Update
image

image
image
image

### CODE REVIEW

**Review by ririxidev — CHANGES_REQUESTED** [review]
image

File: websites/A/Anime Planet/presence.ts [hunk 1]
```diff
@@ -42,15 +44,27 @@ presence.on("iFrameData", (data: IFrameData) => {
...
-  } else if (oldLang !== newLang) {
+  } else if (oldLang !== newLang){
     oldLang = newLang;
     strings = getStrings();
   }
 
+  if (document.location.href !== oldHref){
+    oldHref = document.location.href;
+    oldSettings = false;
+    presence.showSetting(["AnimeDetails", "AnimeState", "MangaDetails", "MangaState"]);
```
Comment (ririxidev) [inline websites/A/Anime Planet/presence.ts:65]: this line.

File: websites/A/Anime Planet/presence.ts [hunk 2]
```diff
@@ -60,182 +74,161 @@ presence.on("UpdateData", async () => {
...
+    if (path.match(key)){
+        presenceData = { ...presenceData, ...value};
+        break;
+    }
+  }
+
+  if (path.includes("/videos/")){
+
+    if (!oldSettings){
+      presence.hideSetting(["MangaDetails", "MangaState"]);
```
Comment (ririxidev) [inline websites/A/Anime Planet/presence.ts:203]: this line.

**Comment by SlowLife1661** [comment]
I made it show settings instead kekw

File: websites/A/Anime Planet/presence.ts [hunk 3]
```diff
@@ -244,7 +237,7 @@ presence.on("UpdateData", async () => {
         },
         {
           label: (await strings).viewSeries,
-          url: document.querySelector<EHref>("h2.sub > a").href
+          url: document.querySelector<ElementHref>('h2.sub > a').href
```
Comment (ririxidev) [inline websites/A/Anime Planet/presence.ts:240]: Already told you to change it.
image

File: websites/A/Anime Planet/presence.ts [hunk 4]
```diff
@@ -256,14 +249,22 @@ presence.on("UpdateData", async () => {
...
-    content.episode.ep = document
-      .querySelector("h1")
-      .textContent.replace(content.title, "")
-      .match(/[1-9]?[0-9]?[0-9]?.?[1-9]?[0-9]?[0-9]/g)[0];
+  } else if (path.includes("/chapters/")){
 
-    presenceData.details = content.title;
-    presenceData.state = `${(await strings).chapter} ${content.episode.ep}`;
+    if (!oldSettings){
+      presence.hideSetting(["AnimeDetails", "AnimeState"]);
```
Comment (ririxidev) [inline websites/A/Anime Planet/presence.ts:255]: this line.

**Comment by ririxidev** [comment]
@SlowLife1661,
But it is not necessary to do it. So literally, remove lines where you do hideSetting/showSetting.

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

File: websites/A/Anime Planet/presence.ts (hunk 3 above)
Comment (SlowLife1661) [reply websites/A/Anime Planet/presence.ts:240]: image

**Comment by SlowLife1661** [comment]
> @SlowLife1661,
> But it is not necessary to do it. So literally, remove lines where you do hideSetting/showSetting.
> 
> Thanks,
> **ririxi**. PreMiD Reviewer Team.
> Contact: [mail@ririxi.dev](mailto:mail@ririxi.dev), ririxi#2721

All done check, my latest comment

File: websites/A/Anime Planet/presence.ts (hunk 3 above)
Comment (ririxidev) [reply websites/A/Anime Planet/presence.ts:227]: ekhem, ``?.href``

Comment (ririxidev) [reply websites/A/Anime Planet/presence.ts:227]: or use ``HTMLHyperlinkElementUtils`` type.

Comment (SlowLife1661) [reply websites/A/Anime Planet/presence.ts:227]: Still the same

Comment (SlowLife1661) [reply websites/A/Anime Planet/presence.ts:227]: It's not like I want to use custom type also lol

Comment (SlowLife1661) [reply websites/A/Anime Planet/presence.ts:227]: image

Comment (SlowLife1661) [reply websites/A/Anime Planet/presence.ts:227]: I hates it when Github comments isn't real-time reee

Comment (SlowLife1661) [reply websites/A/Anime Planet/presence.ts:227]: Ok it works with `HTMLAnchorElement`

Comment (ririxidev) [reply websites/A/Anime Planet/presence.ts]: gud 👍

**Review by ririxidev — APPROVED** [review]
LGTM!

**Review by StrikerFRFX — APPROVED** [review]

**Pull request merged**